src/main/java/com/scylladb/alternator/StreamsAdapterDemo.java

```diff
@@ -0,0 +1,226 @@
+            }
+        }
+
+        LOGGER.info("Starting demo...");
+
+        String srcTable = tablePrefix;
+        String destTable = tablePrefix + "-dest";
+
+        IRecordProcessorFactory recordProcessorFactory = new StreamsRecordProcessorFactory(b, destTable);
+
+        AmazonDynamoDBStreamsAdapterClient adapterClient = new AmazonDynamoDBStreamsAdapterClient(sb.build());
+        AmazonDynamoDB dynamoDBClient = b.build();
+
+        if (ns.getBoolean("cloudwatch")) {
+            cloudWatchClient = AmazonCloudWatchClientBuilder.standard().withCredentials(b.getCredentials())
+                    .withClientConfiguration(b.getClientConfiguration()).build();
+        }
+
+        ExecutorService xs = Executors.newWorkStealingPool(threads);
+
+        try {
+            String streamArn = setUpTables(dynamoDBClient, tablePrefix);
+            ScanResult sr = scanTable(dynamoDBClient, srcTable);
+            ScanResult dr = null;
+
+            if (sr.getCount() < keyNumber && !create_data) {
+                throw new IllegalArgumentException("Not enougn data (" + sr.getCount() + ") in source table");
+            }
+            if (sr.getCount() < keyNumber && create_data) {
+                LOGGER.info("Adding " + (sr.getCount() - keyNumber) + " records to source table...");
+                putItems(dynamoDBClient, tablePrefix, sr.getCount(), keyNumber);
+            }
+
+            KinesisClientLibConfiguration workerConfig = new KinesisClientLibConfiguration("streams-adapter-demo",
+                    streamArn, b.getCredentials(), "streams-demo-worker").withParentShardPollIntervalMillis(1000)
+                            .withCleanupLeasesUponShardCompletion(true).withFailoverTimeMillis(240000)
+                            .withRetryGetRecordsInSeconds(10).withInitialPositionInStream(TRIM_HORIZON)
+                            .withIdleTimeBetweenReadsInMillis(1).withIdleMillisBetweenCalls(1);
+
+            LOGGER.info("Creating worker for stream: " + streamArn);
+            Worker worker = new Worker.Builder().recordProcessorFactory(recordProcessorFactory).config(workerConfig)
+                    .kinesisClient(adapterClient).dynamoDBClient(dynamoDBClient).cloudWatchClient(cloudWatchClient)
+                    .execService(xs).build();
+
+            LOGGER.info("Starting worker...");
+            Thread t = new Thread(worker);
+            t.start();
+
+            for (;;) {
+                Thread.sleep(10000);
+                dr = scanTable(dynamoDBClient, destTable);
+                LOGGER.info("keys synced: " + dr.getCount() + "/" + keyNumber);
+                if (!dr.getCount().equals(keyNumber)) {
+                    continue;
+                }
+                if (dr.getCount() >= keyNumber) {
+                    break;
+                }
+            }
+
+            worker.shutdown();
+            t.join();
+
+            if (dr != null && sr.getItems().equals(dr.getItems())) {
```

Comment:
I think I would like this part optional, i.e. for big scale of using this one, I don't want this one to do this check, I would use or ycsb or the same tool CDC are using.